src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -650,6 +653,66 @@ private boolean matches(Offer offer, Request request) {
         }
     }
 
+    /**
+     * Retrieves the slaveAttributes corresponding to label name. In the event of a match on label names,
+     * returns attributes for SlaveInfo marked as default, otherwiese the matched label's slave attributes. Keeping with
+     * the previous behavior, will return null/empty slave attributes.
+     * @param labelName MesosSlaveInfo label string name.
+     * @return slaveAttributes as a JSONObject.
+     */
+
+    private JSONObject getSlaveAttributeForLabel(String labelName) {
+        LOGGER.info(String.format("Getting slave attributes for label: %s", labelName));
+        List<MesosSlaveInfo> matchedSlaves = getMesosCloud().getSlaveInfos().stream()
+                .filter(slaveInfo -> slaveInfo.getLabelString() != null && slaveInfo.getLabelString().equals(labelName))
```

Comment:
```suggestion
                .filter(slaveInfo -> labelName.equals(slaveInfo.getLabelString()))
```
should avoid the null pointer exception.